swift/Arrow/Sources/Arrow/ArrowCImporter.swift

```diff
@@ -0,0 +1,144 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import ArrowC
+
+public class ArrowCImporter {
+    private func appendToBuffer(
+        _ cBuffer: UnsafeRawPointer?,
+        arrowBuffers: inout [ArrowBuffer],
+        byteCount: Int,
+        length: UInt,
```

Comment:
Why those two parameters? The C Data Interface does not give you any distinct length/capacity information.

Reply:
These are required by the ArrowBuffer.  Length is coming from ArrowArray.length and byteCount is determined from the data.  These values are required by the ArrowBuffer.

Reply:
I understand that the ArrowBuffer constructor takes them. They are just not required as separate arguments for _this function_ :-) It could take a single length argument and pass it for both ArrowBuffer's length and capacity.

(also I don't understand why you are mixing signed and unsigned here)

Reply:
The length and capacity are different right? Length is the amount of elements that are in the array and capacity is the amount of bytes for the buffer so theses are not the same

(the sign issue does need to be resolved.  I plan to resolve in a follow up PR)

Reply:
They are different in general, but when ingesting data from the C Data Interface the only information you can compute is the length. You have to make up a dummy capacity value, and using the length is as good a choice as any :-)

(also, the capacity should only be important for growing or shrinking the buffer, which you hopefully won't attempt on a buffer allocated by another runtime :-))

Reply:
Gotcha, I will update to use only the length. Thank you!